src/features/callAssignments/store.ts

```diff
@@ -30,6 +30,21 @@ const callAssignmentsSlice = createSlice({
         .filter((ca) => ca.id != action.payload.id)
         .concat([action.payload]);
     },
+    callAssignmentUpdated: (
+      state,
+      action: PayloadAction<CallAssignmentData>
+    ) => {
+      const stats = state.statsById[action.payload.id];
+      const callAssignment = state.callAssignments.find(
+        (ca) => ca.id === action.payload.id
+      );
+      if (stats && callAssignment?.cooldown != action.payload.cooldown) {
+        stats.isStale = true;
+        state.callAssignments = state.callAssignments
+          .filter((ca) => ca.id != action.payload.id)
+          .concat([action.payload]);
+      }
```

Comment:
Part of me likes this and thinks the store/reducer is the right place for it. But then another part of me dislikes how this means there's caching logic both in the store and in the model. This needs to be considered (probably by me) as this architecture continues to evolve.